src/helpers.ts

```diff
@@ -0,0 +1,91 @@
+import type { ABI, ABIFunction, ABIInOut } from "./abi.js";
+
+// Regular expression to match tuple types.
+// Example: tuple, tuple[], tuple[5]
+const reTupleType = /^tuple(\[\d*\])?$/;
+
+/**
+ * Fills tuple component's empty names in an ABI with generated names
+ *
+ * @example
+ * Input: {
+ *   "type": "function",
+ *   "selector": "0x95d376d7",
+ *   "payable": false,
+ *   "stateMutability": "payable",
+ *   "inputs": [
+ *     {
+ *       "type": "tuple",
+ *       "name": "",
+ *       "components": [
+ *         { "type": "uint32", "name": "" },
+ *         { "type": "bytes", "name": "" },
+ *         { "type": "bytes32", "name": "" },
+ *         { "type": "uint64", "name": "" },
+ *         { "type": "address", "name": "" }
+ *       ]
+ *     },
+ *     { "type": "bytes", "name": "" }
+ *   ],
+ *   "sig": "assignJob((uint32,bytes,bytes32,uint64,address),bytes)",
+ *   "name": "assignJob",
+ *   "constant": false
+ * }
+ *
+ * Output: {
+ *   "type": "function",
+ *   "selector": "0x95d376d7",
+ *   "payable": false,
+ *   "stateMutability": "payable",
+ *   "inputs": [
+ *     {
+ *       "type": "tuple",
+ *       "name": "",
+ *       "components": [
+ *         { "type": "uint32", "name": "field0" },
+ *         { "type": "bytes", "name": "field1" },
+ *         { "type": "bytes32", "name": "field2" },
+ *         { "type": "uint64", "name": "field3" },
+ *         { "type": "address", "name": "field4" }
+ *       ]
+ *     },
+ *     { "type": "bytes", "name": "" }
+ *   ],
+ *   "sig": "assignJob((uint32,bytes,bytes32,uint64,address),bytes)",
+ *   "name": "assignJob",
+ *   "constant": false
+ * }
+ *
+ * @param abi The ABI to process
+ * @returns A new ABI with tuple component names filled
+ */
+export function abiFillEmptyNames(abi: ABI): ABI {
+  function processComponents(components: ABIInOut[]): void {
+    components.forEach((component, index) => {
+      component.name ||= `field${index}`;
```

Comment:
Should probably refer to them as "parameter" rather than "field? per https://docs.soliditylang.org/en/latest/abi-spec.html

Is there a convention to hint that it's not an official name? Underscore-prefix or something?

I liked your original idea of just doing `_{index}` because it "looks" more generative, but I could be convinced for something else.

Reply:
How about `_p{index}`? We initially tried `_{index}`, but geth requires at least one alphabetical character.

Reply:
Ah. How about `_param{index}` then?

Reply:
I’ve updated it to `_p{index}` with commit add0386da903f63f1bb25538f516c293f534f655

Reply:
@shazow @yohamta I think `param` is okay, but if we're going by the [ABI spec](https://docs.soliditylang.org/en/latest/abi-spec.html) that shazow shared above, wouldn't it be `component` for tuples and not `parameter`?

Reply:
I'm just going by this:

> `name:` the name of the parameter.

Reply:
Yes, you're right. Since we're dealing with components, using `_comp{index}` might make sense too, I think.